Tests/scripts/common.py

```diff
@@ -260,67 +265,104 @@ def person_in_charge(commit):
+def are_pipelines_in_order(current_pipeline:ProjectPipeline, previous_pipeline: ProjectPipeline)-> bool:
     """
-    This function checks if the commit that triggered the current pipeline was pushed
-    after the commit that triggered the the previous pipeline,
-    to avoid rare condition that pipelines are not in the same order as their commits.
+    This function checks if the current pipeline was created after the previous pipeline, to avoid rare conditions 
+    that pipelines are not in the same order as the commits.
     Args:
-        commits: list of gitlab commits
-        current_pipeline_sha: string
-        previous_pipeline_sha: string
+        current_pipeline: The current pipeline object.
+        previous_pipeline: The previous pipeline object.
+    Returns:
+        bool
+    """
+    
+    previous_pipeline_timestamp = parser.parse(previous_pipeline.created_at)
+    current_pipeline_timestamp = parser.parse(current_pipeline.created_at)
+    return current_pipeline_timestamp > previous_pipeline_timestamp
+
 
+def is_pivot(current_pipeline: ProjectPipeline, previous_pipeline: ProjectPipeline)-> bool | None:
+    """
+    Is the current pipeline status a pivot from the previous pipeline status.
+    Args:
+        current_pipeline: The current pipeline object.
+        previous_pipeline: The previous pipeline object.   
     Returns:
-        boolean , the commit that triggered the current pipeline
+        True status changed from success to failed
+        False if the status changed from failed to success
+        None if the status didn't change or the pipelines are not in order of commits
     """
-    current_pipeline_commit_timestamp = None
-    previous_pipeline_commit_timestamp = None
-    the_triggering_commit = None
-    for commit in commits:
-        if commit.id == previous_pipeline_sha:
-            previous_pipeline_commit_timestamp = parser.parse(commit.created_at)
-        if commit.id == current_pipeline_sha:
-            current_pipeline_commit_timestamp = parser.parse(commit.created_at)
-            the_triggering_commit = commit
-    if not current_pipeline_commit_timestamp or not previous_pipeline_commit_timestamp:
-        return False, None
-    return current_pipeline_commit_timestamp > previous_pipeline_commit_timestamp, the_triggering_commit
-
-
-def is_pivot(single_pipeline_id, list_of_pipelines, commits):
+
+    in_order = are_pipelines_in_order(current_pipeline, previous_pipeline)
+    if in_order:
+        if previous_pipeline.status == 'success' and current_pipeline.status == 'failed':
+            return True
+        if previous_pipeline.status == 'failed' and current_pipeline.status == 'success':
+            return False
+    return None
+
+
+def get_reviewer(pr_url: str) -> str | None:
+    approved_reviewer = None
+    try:
+        # Extract the owner, repo, and pull request number from the URL
+        _1, _2,_3, repo_owner, repo_name,_4, pr_number = pr_url.split("/")  
```

Comment:
if some return variable isn't used, you can always write `_` for the return value.
```suggestion
        _ , _ ,_ , repo_owner, repo_name,_ , pr_number = pr_url.split("/")  
```

Reply:
fixed